### PR TITLE
Fix: address failing tests and remove Version3 opt-in since it's required now

### PR DESCRIPTION
**Pull request by cmars**
Mostly onion v2-related. v2 hidden services are no longer supported by
Tor and are considered insecure.

### CODE REVIEW

File: tests/control_cmd_hiddenservice_test.go [hunk 1]
```diff
@@ -2,13 +2,14 @@ package tests
...
 import (
 	"context"
-	"strings"
 	"testing"
 	"time"
 
 	"github.com/cretz/bine/control"
 )
 
+const hsFetchOnion = "2gzyxa5ihm7nsggfxnu52rck2vv4rvmdlkiu3zzui5du4xyclen53wid"
```
Comment (cmars) [inline tests/control_cmd_hiddenservice_test.go:11]: Tor project's onion

File: tests/control_cmd_hiddenservice_test.go [hunk 2]
```diff
@@ -32,7 +33,6 @@ func TestHSFetch(t *testing.T) {
 		ctx.Require.NoError(err)
 	case event := <-eventCh:
 		hsEvent := event.(*control.HSDescContentEvent)
-		ctx.Require.Equal("facebookcorewwwi", hsEvent.Address)
-		ctx.Require.True(strings.HasPrefix(hsEvent.Descriptor, "rendezvous-service-descriptor "+hsEvent.DescID))
```
Comment (cmars) [inline tests/control_cmd_hiddenservice_test.go:36]: Not sure what the v3 equivalent is here. The event content is very different.

File: tor/listen.go [hunk 3]
```diff
@@ -116,6 +116,9 @@ type ListenConf struct {
 // If conf is nil, the default struct value is used. Note, if this errors, any
 // listeners created here are closed but if a LocalListener is provided it may remain open.
 func (t *Tor) Listen(ctx context.Context, conf *ListenConf) (*OnionService, error) {
+	if !conf.Version3 {
+		return nil, fmt.Errorf("v2 onion services no longer supported")
```
Comment (cretz) [inline tor/listen.go:120]: What version are v2 hidden services removed on? Or is it the directory servers and what not that do not support them either? I don't want to break people using old Tor versions if it still works.

Also, if they are absolutely not supported anymore, I'd be ok just removing the field and all v2 related code and always assuming true.

Comment (cretz) [reply tor/listen.go:120]: Ok, I see they are basically completely down on any Tor version. I think we should just remove this config option. I think a backwards incompatible change is acceptable here for removal of a field that is no longer respected.

**Comment by cmars** [comment]
Thanks! A bit of unfinished business here though; I found that `ClientAuths` needs a bit of an overhaul for V3 support as well. I could land that in a followup, but I was leaning toward "everything works on V3" with this PR and make a clean break.

In V3, I think ClientAuths would be expressed as a slice of base32 ed25519 public keys rather than a map -- there's no longer a notion of "name:key", just the key, IIRC the [ADD_ONION command](https://gitweb.torproject.org/torspec.git/tree/control-spec.txt). So this would be another breaking change I think. May have this work pushed up later this evening.

Or I'm happy to land this and follow up, your call!

**Comment by cretz** [comment]
I'll try to take another peek tomorrow on the `ADD_ONION` call because it is frustrating they would make backwards incompatible changes to the protocol spec.

**Comment by cmars** [comment]
Updated with ClientAuthV3 support. It's not a backwards incompatible change from the protocol perspective, just from a bine API perspective. Given that v2 is sunset, perhaps this is OK?

Client auth didn't really have test coverage (unless I'm missing it) -- but I would like to add it.

So we could:
- Land this and follow up with client auth tests
- Revert the client auth stuff here, to cherry-picking into a followup with tests
- Something else? What do you think?

**Comment by cretz** [comment]
@cmars - Hrmm, that makes sense. Unfortunately https://github.com/torproject/torspec/blob/main/control-spec.txt still shows `ClientAuth` and v2.

> Client auth didn't really have test coverage (unless I'm missing it) -- but I would like to add it.

:+1:

> Land this and follow up with client auth tests

I think this is fine. Anything else you want to add before I squash and merge?

**Pull request merged**